src/screens/VisualTests/VisualTests.tsx

```diff
@@ -193,14 +196,19 @@ export const VisualTests = ({
     [reviewTest]
   );
 
+  const build = getFragment(FragmentBuildFields, data?.build || data?.project?.lastBuild);
+
   useEffect(() => {
-    if (isRunning && data?.build && "result" in data.build) {
-      setIsOutdated(false);
-      setIsRunning(false);
+    if (!build) return;
+
+    if ("tests" in build) {
+      updateBuildStatus(build);
     }
```

Comment:
Is this the reason you added the memoization just above?

A simple check here, would save complexity and improve performance.
Sure we can memoize and make the callback do nothing.. but what if we just didn't call the setState when we know it's not needed?

Reply:
Either way we need to check if the build is different to last time, I think it's going to be faster to stringify the status update than the whole build. The alternative would maybe be to look at which tests have changed since the last time we considered updating, but I'm not sure if it's worth it without seeing perf issues.

Reply:
> Is this the reason you added the memoization just above?

I added the memoization to avoid a set state loop -- each time the `VisualTests` component is rendered, it otherwise calls `setStatus` which changes the state of the manager, forcing a re-render of the `VisualTests` component.